bug: OIIO 3.1 does not build on OpenBSD

```
[4/235] /home/brad/ports/pobj/openimageio-3.1.9.0/bin/c++ -DOIIO_INTERNAL=1 -DOpenImageIO_EXPORTS -DOpenImageIO_Util_EXPORTS -DUSE_EXTERNAL_PUGIXML=1 -DUSE_FREETYPE=1 -DUSE_JPEG_TURBO=1 -DUSE_JXL=1 -I/home/brad/ports/pobj/openimageio-3.1.9.0/build-aarch64/include/OpenImageIO -I/home/brad/ports/pobj/openimageio-3.1.9.0/build-aarch64/include -I/home/brad/ports/pobj/openimageio-3.1.9.0/build-aarch64/src/include -I/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include -I/home/brad/ports/pobj/openimageio-3.1.9.0/build-aarch64/deps/include -isystem /usr/local/include -isystem /usr/local/include/Imath -O2 -pipe -DNDEBUG -std=c++17 -fPIC -fvisibility=hidden -Wall -Wformat -Wformat=2 -Wno-unused-function -Wno-overloaded-virtual -Wno-unneeded-internal-declaration -Wno-unused-private-field -Wno-tautological-compare -Qunused-arguments -Wunknown-warning-option -Wno-unused-local-typedefs -Wno-expansion-to-defined -fno-math-errno -MD -MT src/libutil/CMakeFiles/OpenImageIO_Util.dir/argparse.cpp.o -MF src/libutil/CMakeFiles/OpenImageIO_Util.dir/argparse.cpp.o.d -o src/libutil/CMakeFiles/OpenImageIO_Util.dir/argparse.cpp.o -c /home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/libutil/argparse.cpp
FAILED: src/libutil/CMakeFiles/OpenImageIO_Util.dir/argparse.cpp.o
/home/brad/ports/pobj/openimageio-3.1.9.0/bin/c++ -DOIIO_INTERNAL=1 -DOpenImageIO_EXPORTS -DOpenImageIO_Util_EXPORTS -DUSE_EXTERNAL_PUGIXML=1 -DUSE_FREETYPE=1 -DUSE_JPEG_TURBO=1 -DUSE_JXL=1 -I/home/brad/ports/pobj/openimageio-3.1.9.0/build-aarch64/include/OpenImageIO -I/home/brad/ports/pobj/openimageio-3.1.9.0/build-aarch64/include -I/home/brad/ports/pobj/openimageio-3.1.9.0/build-aarch64/src/include -I/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include -I/home/brad/ports/pobj/openimageio-3.1.9.0/build-aarch64/deps/include -isystem /usr/local/include -isystem /usr/local/include/Imath -O2 -pipe -DNDEBUG -std=c++17 -fPIC -fvisibility=hidden -Wall -Wformat -Wformat=2 -Wno-unused-function -Wno-overloaded-virtual -Wno-unneeded-internal-declaration -Wno-unused-private-field -Wno-tautological-compare -Qunused-arguments -Wunknown-warning-option -Wno-unused-local-typedefs -Wno-expansion-to-defined -fno-math-errno -MD -MT src/libutil/CMakeFiles/OpenImageIO_Util.dir/argparse.cpp.o -MF src/libutil/CMakeFiles/OpenImageIO_Util.dir/argparse.cpp.o.d -o src/libutil/CMakeFiles/OpenImageIO_Util.dir/argparse.cpp.o -c /home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/libutil/argparse.cpp
In file included from /home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/libutil/argparse.cpp:16:
In file included from /home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/argparse.h:20:
In file included from /home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/paramlist.h:17:
In file included from /home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/attrdelegate.h:13:
/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/typedesc.h:415:15: error: static assertion failed due to requirement '!std::is_same_v<unsigned long long, unsigned long long>'
  415 | static_assert(!std::is_same_v<unsigned long long, uint64_t>);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/typedesc.h:416:15: error: static assertion failed due to requirement '!std::is_same_v<long long, long long>'
  416 | static_assert(!std::is_same_v<long long, int64_t>);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/typedesc.h:417:19: error: redefinition of 'BaseTypeFromC<unsigned long long>'
  417 | template<> struct BaseTypeFromC<unsigned long long> { static constexpr TypeDesc::BASETYPE value = TypeDesc::UINT64; };
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/typedesc.h:408:19: note: previous definition is here
  408 | template<> struct BaseTypeFromC<uint64_t> { static constexpr TypeDesc::BASETYPE value = TypeDesc::UINT64; };
      |                   ^
/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/typedesc.h:418:19: error: redefinition of 'BaseTypeFromC<const unsigned long long>'
  418 | template<> struct BaseTypeFromC<const unsigned long long> { static constexpr TypeDesc::BASETYPE value = TypeDesc::UINT64; };
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/typedesc.h:409:19: note: previous definition is here
  409 | template<> struct BaseTypeFromC<const uint64_t> { static constexpr TypeDesc::BASETYPE value = TypeDesc::UINT64; };
      |                   ^
/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/typedesc.h:419:19: error: redefinition of 'BaseTypeFromC<long long>'
  419 | template<> struct BaseTypeFromC<long long> { static constexpr TypeDesc::BASETYPE value = TypeDesc::INT64; };
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~
/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/typedesc.h:410:19: note: previous definition is here
  410 | template<> struct BaseTypeFromC<int64_t> { static constexpr TypeDesc::BASETYPE value = TypeDesc::INT64; };
      |                   ^
/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/typedesc.h:420:19: error: redefinition of 'BaseTypeFromC<const long long>'
  420 | template<> struct BaseTypeFromC<const long long> { static constexpr TypeDesc::BASETYPE value = TypeDesc::INT64; };
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/brad/ports/pobj/openimageio-3.1.9.0/OpenImageIO-3.1.9.0/src/include/OpenImageIO/typedesc.h:411:19: note: previous definition is here
  411 | template<> struct BaseTypeFromC<const int64_t> { static constexpr TypeDesc::BASETYPE value = TypeDesc::INT64; };
      |                   ^
6 errors generated.
ninja: build stopped: subcommand failed.
```

OIIO 3.1 does not build on OpenBSD. 3.0 and previous releases are fine.

--- a/src/include/OpenImageIO/typedesc.h
+++ b/src/include/OpenImageIO/typedesc.h
@@ -409,7 +409,7 @@ template<> struct BaseTypeFromC<uint64_t> { static constexpr TypeDesc::BASETYPE 
 template<> struct BaseTypeFromC<const uint64_t> { static constexpr TypeDesc::BASETYPE value = TypeDesc::UINT64; };
 template<> struct BaseTypeFromC<int64_t> { static constexpr TypeDesc::BASETYPE value = TypeDesc::INT64; };
 template<> struct BaseTypeFromC<const int64_t> { static constexpr TypeDesc::BASETYPE value = TypeDesc::INT64; };
-#if defined(__GNUC__) && (ULONG_MAX == 0xffffffffffffffff) && !(defined(__APPLE__) && defined(__MACH__)) || defined(__NetBSD__)
+#if defined(__GNUC__) && (ULONG_MAX == 0xffffffffffffffff) && !(defined(__APPLE__) && defined(__MACH__)) && !defined(__OpenBSD__) || defined(__NetBSD__)
 // Some platforms consider int64_t and long long to be different types, even
 // though they are actually the same size.
 static_assert(!std::is_same_v<unsigned long long, uint64_t>);

